enginetest/spatial_index_tests.go

```diff
@@ -0,0 +1,453 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	sqle "github.com/dolthub/go-mysql-server"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+type SpatialIndexPlanTestAssertion struct {
+	q        string
+	skip     bool
+	skipPrep bool
+	noIdx    bool
+	exp      []sql.Row
+}
+
+type SpatialIndexPlanTest struct {
+	name  string
+	setup []string
+	tests []SpatialIndexPlanTestAssertion
+}
+
+var SpatialIndexTests = []SpatialIndexPlanTest{
```

Comment:
i don't see any tests with compound filters -- AND, OR, etc. How do we choose between two choices for filtered index access? For two spatial options, but also a spatial/regular index choice. Can we intersect (AND) two spatials/spatial+regular? Similar treatment w/ concat nodes and OR? Maybe a test with filters that should be pushed into subqueries too, like `select st_aswkt(g) from (select * from geom_tbl) t where st_intersects(g, point(0,0))`